buildingmotif/label_parsing/SerializedParserMetrics.py

```diff
@@ -0,0 +1,209 @@
+import importlib.util
+import os
+import re
+
+from buildingmotif.label_parsing.build_parser import (
+    generate_parsers_for_clusters,
+    generate_parsers_for_points,
+)
+from buildingmotif.label_parsing.combinators import (
+    COMMON_EQUIP_ABBREVIATIONS_BRICK,
+    COMMON_POINT_ABBREVIATIONS,
+)
+from buildingmotif.label_parsing.tools import abbreviationsTool, codeLinter
+
+
+class SerializedParserMetrics:
+    """
+    Combines parsers into a compact class with detailed metrics and other information.
+    Allows for easier serialization.
+
+    Attributes:
+        parsers(List[str]): list of all parsers
+        serializers_list (List[Dict]): list of all serialized parsers
+        clusters(List[List[str]]): list of all clusters (clusters can have only one element)
+        distance_metrics(Dict): statistics for distance matrix with similarity ratio as distance metric (mean, median, std, min, max, range)
+        clustering_metrics(Dict): statistics for clustering (number of clusters, noise points, and silhouette score)
+        flagged_abbreviations(List): LLM-flagged abbreviations
+        list_of_dicts(List[Dict]): list of dictionaries with abbreviations matched to brick classes. defaults to COMMON_EQUIP_ABBREVIATIONS_BRICK, COMMON_POINT_ABBREVIATIONS
+        parsed_count(int): total parsed across all clusters
+        unparsed_count(int): total unparsed across all clusters
+        total_count(int): total in all clusters
+
+        combined_clusters:List[dict]:
+        for each cluster, each Dict has
+        -parser(Dict): serialized parser
+        -source_code(str): parser code
+        -tokens(List): emitted tokens from running parser on cluster
+        -parsed_labels(List): building point labels in which parser did not contain an error
+        -unparsed(List): building point labels in which parser contained an error
+        -parser_metrics(Dict):
+            -parsed_count(int): count of parsed for that cluster
+            -unparsed_count(int): count of unparsed for that cluster
+            -total_count(int): count of total for that cluster
+    """
+
+    def __init__(
+        self,
+        filename: str,
+        col_name: str,
+        num_tries=3,
+        list_of_dicts=[COMMON_EQUIP_ABBREVIATIONS_BRICK, COMMON_POINT_ABBREVIATIONS],
+    ):
+        """
+        Initializes a new SerializedParserMetrics object.
+
+        Args:
+            filename (str): file path to csv file
+            col_name (str): relevant column where data is stored
+            num_tries (Optional, int): max number of times for LLM to try to generate LLM_Token_Predictions object. Defaults to 3.
+            list_of_dicts (Optional, List): List of dictionaries where abbreviation is matched to brick class. Defaults to [equip_abbreviations, point_abbreviations]
+        """
```

Comment:
thanks for putting in this documentation! Can you change the style to match what we are using elsewhere in the project? See [this](https://github.com/NREL/BuildingMOTIF/blob/develop/buildingmotif/dataclasses/template.py#L221C1-L225C25) for an example. This kind of translation task is a great use of genAI